gobblin-cluster/src/main/java/org/apache/gobblin/cluster/SleepingTask.java

```diff
@@ -17,24 +17,51 @@
 
 package org.apache.gobblin.cluster;
 
+import java.io.IOException;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
 import avro.shaded.com.google.common.base.Throwables;
 import lombok.extern.slf4j.Slf4j;
 
 import org.apache.gobblin.runtime.TaskContext;
+import org.apache.gobblin.runtime.TaskState;
 import org.apache.gobblin.runtime.task.BaseAbstractTask;
 
 @Slf4j
 public class SleepingTask extends BaseAbstractTask {
+  public static final String TASK_STATE_FILE_KEY = "task.state.file.path";
+
   private final long sleepTime;
+  private FileSystem fs;
```

Comment:
Do you need Hadoop FileSystem here? If you use java filesystem, you can set it to delete on process shutdown to ensure cleanup happens.

Reply:
Used Java File API.